Implement scheme for custom arm_pass_manager insertions (#16396)

3l1 opened the pull request:
Summary:


Implement scheme for custom arm_pass_manager + ...


bypass-github-export-checks
bypass-github-pytorch-ci-checks
bypass-github-executorch-ci-checks

Reviewed By: digantdesai, Ninja91

Differential Revision: D87120925

digantdesai reviewed (approved):
Review automatically exported from Phabricator review in Meta.

per commented:
Could you add some tests to exercise this code as well? And show how the pass manager subclass is to be replaced in the backend in a non-intrusive way?

zingo commented:
Hi I see you linkrunner error is 

    291  295 |     def _configure_pass_insertions(self, exported_program: ExportedProgram) -> None:
    291      |-        """Hook for subclasses to configure pass insertions.
    292      |-Called at the STARTof pipeline construction, before any passes are added.
         296 |+        """Hook for subclasses to configure pass insertions.Called at the START
         297 |+of pipeline construction, before any passes are added.
    294  298 | 

I think this was fixed recently You probably need to rebase it. Ill try to rebase rebasing it here to check but you might get meta internal mismatch, let see if so you need to do it on oyr side or re-sync internal version.

3l1 committed:
Implement scheme for custom arm_pass_manager insertions (#16396)

Summary:


Implement scheme for custom arm_pass_manager + ...


bypass-github-export-checks
bypass-github-pytorch-ci-checks
bypass-github-executorch-ci-checks

Reviewed By: digantdesai, Ninja91

Differential Revision: D87120925
